libcli.c

```diff
@@ -1966,11 +1990,1146 @@ void cli_set_idle_timeout_callback(struct cli_def *cli, unsigned int seconds, in
 void cli_telnet_protocol(struct cli_def *cli, int telnet_protocol) {
   cli->telnet_protocol = !!telnet_protocol;
 }
-
 void cli_set_context(struct cli_def *cli, void *context) {
   cli->user_context = context;
 }
 
 void *cli_get_context(struct cli_def *cli) {
   return cli->user_context;
 }
+
+struct cli_command *cli_register_filter(struct cli_def *cli, const char *command,
+                                        int (*init)(struct cli_def *cli, int, char **, struct cli_filter *),
+                                        int (*filter)(struct cli_def *, const char *, void *), int privilege, int mode,
+                                        const char *help) {
+  struct cli_command *c;
+
+  if (!command) return NULL;
+  if (!(c = calloc(sizeof(struct cli_command), 1))) return NULL;
+
+  c->command_type = CLI_FILTER_COMMAND;
+  c->init = init;
+  c->filter = filter;
+  c->next = NULL;
+  if (!(c->command = strdup(command))) {
+    free(c);
+    return NULL;
+  }
+
+  c->privilege = privilege;
+  c->mode = mode;
+  if (help && !(c->help = strdup(help))) {
+    free(c->command);
+    free(c);
+    return NULL;
+  }
+  // filters are all registered at the top level
+  if (!cli_register_command_core(cli, NULL, c)) {
+    cli_free_command(c);
+    c = NULL;
```

Comment:
`return NULL` to match the previous blocks?

Reply:
Could, but it falls through and returns NULL anyways.